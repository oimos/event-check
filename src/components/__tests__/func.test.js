import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../app';

const testProps = {
  defaultAccoutName: 'ラクテンタロー',
  errorNote: '',
  inputName: 'accountNumber',
  isMobile: false,
  isValid: true,
  labelText: '口座名義',
  notes: '※お申込者ご本人名義をご入力ください',
  placeholder: '例) ラクテンタロウ（セイメイ）',
  updateAccountNumber: () => {},
  name: 'updateAccountNumber',
};

const shallowSetUp = props => shallow(<App {...props} />);
const mountSetUp = props => mount(<App {...props} />, { disableLifecycleMethods: true });

describe('App', () => {
  let shallowComponent;
  let mountComponent;
  beforeEach(() => {
    shallowComponent = additionalProps => shallowSetUp({ ...testProps, ...additionalProps });
    mountComponent = additionalProps => mountSetUp({ ...testProps, ...additionalProps });
  });

  it('snapshot', () => {
    const elem = shallowComponent();
    expect(toJson(elem)).toMatchSnapshot();
  });
});
