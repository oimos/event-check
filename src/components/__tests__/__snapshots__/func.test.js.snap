// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<Fragment>
  <div
    className="outer"
  >
    Test
  </div>
  <div
    className="inner"
  >
    Test
  </div>
</Fragment>
`;
